esmvaltool/diag_scripts/weighting/climwip.py

```diff
@@ -418,19 +439,53 @@ def visualize_and_save_weights(weights: 'xr.DataArray', cfg: dict,
     log_provenance(caption, filename_data, cfg, ancestors)
 
 
+def get_variable_groups(contribution: str, cfg: dict) -> list:
+    """Return a list of variable groups with contributions > 0."""
+    if cfg.get(contribution, None) is not None:
+        return [key for key, value in cfg[contribution].items() if value > 0]
+    return []
+
+
 def main(cfg):
     """Perform climwip weighting method."""
     models, observations = read_metadata(cfg)
 
-    variable_groups = models.keys()
+    variable_groups_independence = get_variable_groups(
+        'independence_contributions', cfg)
+    variable_groups_performance = get_variable_groups(
+        'performance_contributions', cfg)
+
+    if (len(variable_groups_independence) == 0 and
+            len(variable_groups_performance) == 0):
```

Comment:
I think this also works (and is more readable):
```suggestion
    if not (variable_groups_independence or variable_groups_performance):
```

Reply:
And here you could then use `contributions_performance` and `contributions_independence`